genesyscloud/webdeployments_configuration/utils/resource_genesyscloud_webdeployments_configuration_messenger.go

```diff
@@ -79,9 +79,17 @@ func buildMessengerApps(apps []interface{}) *platformclientv2.Messengerapps {
 	}
 
 	app := apps[0].(map[string]interface{})
+	conversations, conversationsOk := app["conversations"].([]interface{})
+	knowledge, knowledgeOk := app["knowledge"].([]interface{})
+
+	if !conversationsOk || !knowledgeOk {
+		fmt.Println("error in build Messenger Apps")
+		return nil
+	}
```

Comment:
I don't think we should return in this case. If `conversations` and `knowledge` are optional fields, then the user should be able to omit them from their tf config, and then they'll appear as nil in the resource schema data and `knowledgeOk` will be `false`. 

I would check `if ok { messengerApp.Conversations = buildAppConversation(conversations) }`

Reply:
Nice catch Charlie